sklearn/utils/estimator_checks.py

```diff
@@ -4162,6 +4162,66 @@ def fit_transform(est):
         assert_allclose_dense_sparse(X_trans_no_setting, X_trans_default)
 
 
+def _output_from_fit_transform(transformer, name, X, df, y):
+    """Generate output to set test `set_output` for different configuration:
+
+    - calling either `fit.transform` or `fit_transform`;
+    - passing either a dataframe or a numpy array to fit;
+    - passing either a dataframe or a numpy array to transform.
+    """
+    outputs = {}
+
+    # fit then transform case:
+    cases = [
+        "fit.transform/fit_df/transform_df",
```

Comment:
I find using `zip` later is confusing. Can we list out the cases here and enumerate over it? For example:

```python
        ("fit.transform/fit_df/transform_df", [df, df, y]),
```

Reply:
done